pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -189,3 +289,77 @@ func (ia *IngressAnalyzer) getRouteTargetedPeers(ns string, svcList []string) []
 	}
 	return res
 }
+
+/*************************************************************************************************/
+// Ingress allowed connections by k8s-ingress objects
+
+// allowedIngressConnectionsByk8sIngress returns map from peers names to the allowed ingress connections based on k8s-Ingress objects rules
+func (ia *IngressAnalyzer) allowedIngressConnectionsByk8sIngress() (map[string]common.Connection, error) {
+	medRes := make(map[string]*common.ConnectionSet)
+	for ns, objSvcMap := range ia.k8sIngressToServicesMap {
+		// if there are no services in same namespace of the Ingress, the ingress objects in this ns will be skipped
+		if _, ok := ia.servicesToPeersMap[ns]; !ok {
+			continue
+		}
+
+		for _, svcList := range objSvcMap {
+			ingressObjTargetPeersAndPorts, err := ia.getIngressObjectTargetedPeersAndPorts(ns, svcList)
+			if err != nil {
+				return nil, err
+			}
+			// avoid duplicates in the result, consider the different ports supported
+			for peer, pConn := range ingressObjTargetPeersAndPorts {
+				if _, ok := medRes[peer.String()]; !ok {
+					medRes[peer.String()] = pConn
+				} else {
+					medRes[peer.String()].Union(pConn)
+				}
+			}
+		}
+	}
+	// convert all conns to common.Connection for outer representation
+	res := make(map[string]common.Connection, len(medRes))
+	for peerStr, pConn := range medRes {
+		res[peerStr] = pConn
+	}
+	return res, nil
+}
+
+// getIngressObjectTargetedPeersAndPorts returns map from peers which are targeted by Ingress objects in their namespace to
+// the Ingress required connections
+func (ia *IngressAnalyzer) getIngressObjectTargetedPeersAndPorts(ns string,
+	svcList []serviceInfo) (map[eval.Peer]*common.ConnectionSet, error) {
+	res := make(map[eval.Peer]*common.ConnectionSet)
+	for _, svc := range svcList {
+		peers, ok := ia.servicesToPeersMap[ns][svc.serviceName]
+		if !ok {
+			ia.logger.Warnf("ignoring target service " + svc.serviceName + " : service not found")
+		}
+		for _, peer := range peers {
+			permittedPeerConn := common.MakeConnectionSet(false)
+			// check if its TCP conns contains the required port
+			peerTCPConn := eval.GetPeerExposedTCPConnections(peer)
+			if peerTCPConn.Contains(svc.serviceTargetPort.String(), string(corev1.ProtocolTCP)) {
+				permittedPort := common.PortSet{}
+				var portNum int
+				if svc.serviceTargetPort.Type == intstr.Int {
+					portNum = svc.serviceTargetPort.IntValue()
+				} else {
+					portInt, err := ia.pe.ConvertPeerNamedPort(svc.serviceTargetPort.StrVal, peer)
+					if err != nil {
+						return nil, err
+					}
+					portNum = int(portInt)
+				}
```

Comment:
do we have a test with k8s Ingress rule with named port ?

Reply:
added